src/lenskit/data/relationships.py

```diff
@@ -163,19 +184,86 @@ def pandas(self, *, attributes: str | list[str] | None = None, ids=False) -> pd.
         return tbl.to_pandas()
 
     def matrix(
-        self, *, combine: MAT_AGG | Mapping[str, MAT_AGG] | None = None
+        self, *, row_entity: str = "user", col_entity: str = "item"
+    ) -> MatrixRelationshipSet:  # pragma: nocover
+        if (row_entity, col_entity) in self.mat_rset:
+            return self.mat_rset[(row_entity, col_entity)]
+        else:
+            new_mat_rset = self._make_matrix(row_entity=row_entity, col_entity=col_entity)
+            self.mat_rset[(row_entity, col_entity)] = new_mat_rset
+            return new_mat_rset
+
+    def _make_matrix(
+        self, *, row_entity: str = "user", col_entity: str = "item"
     ) -> MatrixRelationshipSet:  # pragma: nocover
```

Comment:
We're actually writing this now, and should be testing it, so remove `# pragma: nocover`.